frontend/src/components/visualizations/Audio/Audio.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { RootState } from "../../../types/data";
+import {
+	GENERATE_FILE_URL,
+	generateFileDownloadUrl as generateFileDownloadUrlAction,
+} from "../../../actions/file";
+
+import {
+	GENERATE_VIZ_URL,
+	generateVizDataDownloadUrl as generateVizDataDownloadUrlAction,
+} from "../../../actions/visualization";
+
+type AudioProps = {
+	fileId?: string;
+	visualizationId?: string;
+};
+
+export default function Audio(props: AudioProps) {
+	const { fileId, visualizationId } = props;
+
+	const dispatch = useDispatch();
+
+	const generateFileDownloadUrl = (
```

Comment:
I wonder if there is a way to simplify this set of chained effects and actions. We can chat about this in person.